File: slicing-ios/SlicingGame/Page/Storage/PageCache.swift
//
//  PageCache.swift
//  Page storage: cached loaded page items
//

import Foundation

class PageCache {
    
    // --
    // MARK: Singleton instance
    // --
    
    public static let shared = PageCache()


    // --
    // MARK: Members
    // --
    
    private var entries = [String: Page]()
    

    // --
    // MARK: Cache access
    // --
    
    func hasEntry(cacheKey: String) -> Bool {
        return entries[cacheKey] != nil
    }
    
    func getEntry(cacheKey: String) -> Page? {
        return entries[cacheKey]
    }
    
    func storeEntry(cacheKey: String, page: Page) {
        entries[cacheKey] = page
    }
    
    func removeEntry(cacheKey: String) {
        entries.removeValue(forKey: cacheKey)
    }
    
    func clear() {
        entries = [:]
    }
    
}

File: slicing-android/app/src/main/java/com/crescentflare/slicinggame/page/storage/PageCache.kt
package com.crescentflare.slicinggame.page.storage

/**
 * Page storage: cached loaded page items
 */
object PageCache {

    // --
    // Members
    // --

    private var entries = mutableMapOf<String, Page>()


    // --
    // Cache access
    // --

    fun hasEntry(cacheKey: String): Boolean {
        return entries[cacheKey] != null
    }

    fun getEntry(cacheKey: String): Page? {
        return entries[cacheKey]
    }

    fun storeEntry(cacheKey: String, page: Page) {
        entries[cacheKey] = page
    }

    fun removeEntry(cacheKey: String) {
        entries.remove(cacheKey)
    }

    fun clear() {
        entries = mutableMapOf()
    }

}
